specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+
+## Data Types and Values
+
+Within the Base Profile, defining local variables is not supported; instructions
+cannot be nested and constant expressions are fully evaluated. This implies the
+following:
+
+- Call arguments must be constant values, and `inttoptr` casts as well as
+  `getelementptr` instructions must be inlined into a call instruction.
+- It is not possible to express classical computations as part of a Base Profile
+  compliant program.
+
+Constants of any type are permitted as part of a function call. What data types
+occur in the program hence depends entirely on the used QIS functions, as well
+as the runtime functions used for output recording. Constant values of type
+`i64` in particular may be used as part of calls to output recording functions;
+see the section on [output recording](#output-recording) for more details.
+
+For a program to be considered valid, it must make use of at least one qubit and
+perform at least one measurement. The `%Qubit` and `%Result` data types hence
+must be supported by all backends. Qubits and results can occur only as
+arguments in function calls and are represented as a pointer of type `%Qubit*`
+and `%Result*` respectively, where the pointer itself identifies the qubit or
+result value rather than a memory location where the value is stored: a 64-bit
+integer constant is cast to the appropriate pointer type. A more detailed
+elaboration on the purpose of this representation is given in the next
+subsection. The integer constant that is cast must be in the interval `[0,
+nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits`
+and `nrResults` are the required number of qubits and results defined by the
+corresponding [entry point attributes](#attributes). Since backends may look at
+the values of the `required_qubits` and `required_results` attributes to
+determine whether a program can be executed, it is recommended to index qubits
+and results consecutively such that there are no unused values within these
+ranges.
+
+### Qubit and Result Usage
+
+The amount of available memory on a QPU both with regards to qubits and
+potentially also with regards to classical memory for storing measurement
+results before they are read out and transmitted to another classical processor
+is commonly still fairly limited. Any memory - quantum or classical - that is
+used during quantum execution is not usually managed dynamically. Instead,
+operations are scheduled and resources are bound as part of compilation. How
+early in the process this happens varies, and QIR permits to express programs in
+a form that either defers allocation and management of such resources to later
+stages, or to directly identify individual qubits and results by a constant
+integer value as outlined above. This permits to accurately reflect application
+intent for a variety of frontends.
+
+Ultimately, it is up to the executing backend what data structure is associated
```

Comment:
"what data structure" => "which data structure"?

Reply:
I am not sure - that sounds odd to me, but I am not a native English speaker.

Reply:
Not being either, I checked on Internet and, in summary, grammar websites tend to agree on this:

> Although you can sometimes use “what” and “which” interchangeably, there are certain instances where only one of these words is correct. 
>
> The key differences between “what” and “which” are how many possibilities there are and how much you know about those possibilities. 
>
> Use “what” when there are lots of possible options or when you don’t know how many options there are. 
>
> In contrast, “which” indicates there are fewer possible options. You already know that the answer is one of two things or one of a few things.

Up to you. :smile:

Reply:
Thanks for the language help. :) Updated.